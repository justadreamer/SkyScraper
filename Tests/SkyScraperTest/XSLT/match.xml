<?xml version="1.0"?>
<a>
  <c>n</c>
  <c>no spot no</c>
  <c>nO</c>
  <c>noOo spot stop</c>
  <c>stop nooo</c>
  <c>xnoOox</c>
  <c>non no n</c>
  <c>none</c>
</a>
