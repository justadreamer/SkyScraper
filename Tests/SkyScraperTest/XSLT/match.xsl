<?xml version="1.0"?>
<xsl:stylesheet version="1.0" 
                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                xmlns:regexp="http://exslt.org/regular-expressions"
                extension-element-prefixes="regexp">

<xsl:template match="c">
  <xsl:variable name="val"><xsl:value-of select="."/></xsl:variable>
  regexp: no+ (case)          string: <xsl:value-of select="." /> result: <xsl:for-each select="regexp:match($val, 'no+', '')"><xsl:value-of select="position()" /> = <xsl:value-of select="." />, </xsl:for-each>
  regexp: no+ (nocase)        string: <xsl:value-of select="." /> result: <xsl:for-each select="regexp:match($val, 'no+', 'i')"><xsl:value-of select="position()" /> = <xsl:value-of select="." />, </xsl:for-each>
  regexp: no+ (case,global)   string: <xsl:value-of select="." /> result: <xsl:for-each select="regexp:match($val, 'no+', 'g')"><xsl:value-of select="position()" /> = <xsl:value-of select="." />, </xsl:for-each>
  regexp: no+ (nocase,global) string: <xsl:value-of select="." /> result: <xsl:for-each select="regexp:match($val, 'no+', 'gi')"><xsl:value-of select="position()" /> = <xsl:value-of select="." />, </xsl:for-each>
</xsl:template>
<xsl:template match="emoji">
<xsl:variable name="val"><xsl:value-of select="."/></xsl:variable>
regexp: ðŸŒµðŸŒ· (case)          string: <xsl:value-of select="." /> result: <xsl:for-each select="regexp:match($val, 'ðŸŒµðŸŒ·', '')"><xsl:value-of select="position()" /> = <xsl:value-of select="." />, </xsl:for-each>
</xsl:template>
</xsl:stylesheet>
