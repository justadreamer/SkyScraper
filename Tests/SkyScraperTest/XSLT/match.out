<?xml version="1.0"?>

  
  regexp: no+ (case)          string: n result: 
  regexp: no+ (nocase)        string: n result: 
  regexp: no+ (case,global)   string: n result: 
  regexp: no+ (nocase,global) string: n result: 
  
  regexp: no+ (case)          string: no spot no result: 1 = no, 
  regexp: no+ (nocase)        string: no spot no result: 1 = no, 
  regexp: no+ (case,global)   string: no spot no result: 1 = no, 2 = no, 
  regexp: no+ (nocase,global) string: no spot no result: 1 = no, 2 = no, 
  
  regexp: no+ (case)          string: nO result: 
  regexp: no+ (nocase)        string: nO result: 1 = nO, 
  regexp: no+ (case,global)   string: nO result: 
  regexp: no+ (nocase,global) string: nO result: 1 = nO, 
  
  regexp: no+ (case)          string: noOo spot stop result: 1 = no, 
  regexp: no+ (nocase)        string: noOo spot stop result: 1 = noOo, 
  regexp: no+ (case,global)   string: noOo spot stop result: 1 = no, 
  regexp: no+ (nocase,global) string: noOo spot stop result: 1 = noOo, 
  
  regexp: no+ (case)          string: stop nooo result: 1 = nooo, 
  regexp: no+ (nocase)        string: stop nooo result: 1 = nooo, 
  regexp: no+ (case,global)   string: stop nooo result: 1 = nooo, 
  regexp: no+ (nocase,global) string: stop nooo result: 1 = nooo, 
  
  regexp: no+ (case)          string: xnoOox result: 1 = no, 
  regexp: no+ (nocase)        string: xnoOox result: 1 = noOo, 
  regexp: no+ (case,global)   string: xnoOox result: 1 = no, 
  regexp: no+ (nocase,global) string: xnoOox result: 1 = noOo, 
  
  regexp: no+ (case)          string: non no n result: 1 = no, 
  regexp: no+ (nocase)        string: non no n result: 1 = no, 
  regexp: no+ (case,global)   string: non no n result: 1 = no, 2 = no, 
  regexp: no+ (nocase,global) string: non no n result: 1 = no, 2 = no, 
  
  regexp: no+ (case)          string: none result: 1 = no, 
  regexp: no+ (nocase)        string: none result: 1 = no, 
  regexp: no+ (case,global)   string: none result: 1 = no, 
  regexp: no+ (nocase,global) string: none result: 1 = no, 


